crates/astria-sequencer/src/mempool/mod.rs

```diff
@@ -211,15 +245,28 @@ impl Mempool {
                 );
                 // promote the transactions
                 for ttx in to_promote {
+                    let tx_id = ttx.id();
                     if let Err(error) =
                         pending.add(ttx, current_account_nonce, &current_account_balances)
                     {
+                        // remove from tracked
```

Comment:
If code doing something as simple as this needs to be commented, then its either not self evident (leave a comment tagged with `XXX` or `NOTE`, see below), or adjust the called methods to make it evident.